src/jit/importer.cpp

```diff
@@ -7522,7 +7522,9 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
         }
 
-        if (!bIntrinsicImported)
+        // Don't create cast for a variable on the stack if it is intrinsic or tail call.
+        // We assume that the value will not be physically represented there.
+        if (!bIntrinsicImported && !(tailCall && canTailCall))
```

Comment:
I am not sure this is the right fix for the following reason:

During importation, impImportCall(), some tail call checks are performed and remaining checks during fgMorphCall().  After the checks in fgMorphCall(), if JIT is committed to honor tail call, it will set a flag (GTF_CALL_M_TAILCALL) on GT_CALL node. Till then there is no guarantee a tail call will be honored.

Say we don't insert the cast during importation even if the tail call is returning a small type.  What if the tail call was not honored and it was dispatched as a normal call?  Is there anything to ensure that the cast is added back again?